scylla/src/transport/connection_pool.rs

```diff
@@ -245,6 +263,82 @@ impl NodeConnectionPool {
     }
 }
 
+struct Keepaliver {
+    connections: Arc<ArcSwap<MaybePoolConnections>>,
+    keepalive_interval: Duration,
+}
+
+impl Keepaliver {
+    pub fn load_connections(&self) -> Vec<Arc<Connection>> {
+        use MaybePoolConnections::*;
+        use PoolConnections::*;
+
+        let pool = self.connections.load_full();
+        match &*pool {
+            Ready(NotSharded(conns)) => conns.clone(),
+            Ready(Sharded { connections, .. }) => connections.iter().flatten().cloned().collect(),
+            Initializing => vec![],
+            Broken => {
+                warn!("Collecting connections to keepalive failed, connection pool is broken");
```

Comment:
Just a nitpick: I'm not sure if we should print that. "Broken" means that all previous connections broke and the pool didn't manage to reconnect any of them yet. If you had a per-connection task which runs the keepalives, then all of them would be dropped at this point and wouldn't print anything.

@psarna thoughts?

Reply:
Yes, I think it's redundant. Nodes are sometimes shut down for various reasons (maintenance, before being decommissioned etc.) and I'm not sure this is serious enough to print warnings. I agree we should lower the verbosity (debug sounds good).

Apart from that, I think the message should be improved, too. Maybe it doesn't matter now if we lower the verbosity to debug, but a user sees it I feel they won't understand it (what does "collecting connections to keepalive" or "connection pool being broken" mean to them?). Maybe something clearer would be better, e.g. "Cannot send connection keepalives for node {} as there are no alive connections in the pool"?